src/account.c

```diff
@@ -236,27 +239,55 @@ account_iface_init (XdpDbusAccountIface *iface)
   iface->handle_get_user_information = handle_get_user_information;
 }
 
+static void
+account_dispose (GObject *object)
+{
+  Account *account = (Account *) object;
+
+  g_clear_object (&account->impl);
+
+  G_OBJECT_CLASS (account_parent_class)->dispose (object);
+}
+
 static void
 account_init (Account *account)
 {
-  xdp_dbus_account_set_version (XDP_DBUS_ACCOUNT (account), 1);
 }
 
 static void
 account_class_init (AccountClass *klass)
 {
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->dispose = account_dispose;
+}
+
+static Account *
+account_new (XdpDbusImplAccount *impl)
```

Comment:
Hmm any particular reason to split things out into a separate `*_new` method? Seems like a somewhat arbitrary split